model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/solve_nonhydro_stencils.py

```diff
@@ -618,8 +620,8 @@ def _stencils_42_44_45(
         (z_w_expl, z_contr_w_fl_l),
     )
 
-    (z_beta, z_alpha) = where(
-        (k_field >= 0) & (k_field < nlev),
+    (z_beta, z_alpha) = concat_where(
+        0 <= dims.KDim < nlev,
```

Comment:
```suggestion
        (0 <= dims.KDim) & (dims.KDim < nlev),
```

or simply `dims.KDim < nlev`